### PR TITLE
troubleshooting: mention missing files

### LINKED ISSUE #1660: pull: add troubleshooting hints for failed command
It has been mentioned (https://github.com/iterative/dvc/issues/4316) that the error text when checkout fails after pulling from an out of date remote is not very clear. In attempting to clarify the error (https://github.com/iterative/dvc/pull/4339) it was suggested that a link to a "common issues" page would be helpful.

Looking at the website, I agree that https://dvc.org/doc/user-guide/troubleshooting might be a good place to put this.

### PR DESCRIPTION
**Pull request by bobertlo**
Here is a draft to document a common issue, with the goal to link here from an error message (https://github.com/iterative/dvc/pull/4339) Any feedback is appreciated.

Fixes #1660

> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: content/docs/user-guide/troubleshooting.md [hunk 1]
```diff
@@ -7,6 +7,14 @@ This file uses a special engine feature for the following headers, so that a
 custom anchor link is used. Just add {#custom-anchor} after each title:
 -->
 
+## Failed to pull data from the cloud {#missing-files}
+
+Users may encounter errors when running `dvc pull` and `dvc fetch` like
+`WARNING: Cache 'xxxx' not found.` or `ERROR: failed to pull data from the cloud`.
+When working across multiple systems, the most common cause is changes pushed to a
+git repository without the corresponding data present in the DVC remote. Make sure
```
Comment (shcheklein) [inline content/docs/user-guide/troubleshooting.md:15]: minor: git -> Git

File: content/docs/user-guide/troubleshooting.md [hunk 2]
```diff
@@ -7,6 +7,14 @@ This file uses a special engine feature for the following headers, so that a
...
+to `dvc push` from the remote system and try again.
```
Comment (jorgeorpinel) [inline content/docs/user-guide/troubleshooting.md:16]:
```suggestion
Users may encounter errors when running `dvc pull` and `dvc fetch`, like
`WARNING: Cache 'xxxx' not found.` or `ERROR: failed to pull data from the cloud`.
The most common cause is changes pushed to a
Git repository without the corresponding data present in the DVC remote. Make sure
to `dvc push` from the original project, and try again.
```
^ Needs formatting.

File: content/docs/user-guide/troubleshooting.md [hunk 3]
```diff
@@ -7,6 +7,14 @@ This file uses a special engine feature for the following headers, so that a
 custom anchor link is used. Just add {#custom-anchor} after each title:
 -->
 
+## Failed to pull data from the cloud {#missing-files}
```
Comment (jorgeorpinel) [inline content/docs/user-guide/troubleshooting.md:10]: Will you be adding a link to error.dvc.org/missing-files in iterative/dvc/pull/4339 ?

Comment (jorgeorpinel) [reply content/docs/user-guide/troubleshooting.md:10]: (Please let us know when that PR is approved or merged.)

Comment (bobertlo) [reply content/docs/user-guide/troubleshooting.md:10]: Yes I have that queued right now.

**Comment by jorgeorpinel** [comment]
Actually... We can merge this! No need to wait for core as the changes aren't completely inter-dependent.

**Pull request merged**